mmv1/third_party/terraform/utils/provider.go.erb

```diff
@@ -496,6 +496,12 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Pr
         <% end -%>
 	}
 
+	// opt in extension for adding to the User-Agent header
+	if ext := os.Getenv("GOOGLE_TERRAFORM_USERAGENT_EXTENSION"); ext != "" {
+		ua := config.userAgent
+		config.userAgent = fmt.Sprintf("%s %s", ua, ext)
```

Comment:
nit: you can just inline `config.userAgent` in for `ua`

Reply:
noop for readability imo